frame/contracts/src/tests.rs

```diff
@@ -300,17 +281,13 @@ fn compile_module<T>(wabt_module: &str)
 	Ok((wasm, code_hash))
 }
 
-// Perform a simple transfer to a non-existent account supplying way more gas than needed.
-// Then we check that the all unused gas is refunded.
+/// The payment for gas is handled outside of contract execution (e.g `call`) functions.
 #[test]
-fn refunds_unused_gas() {
-	ExtBuilder::default().gas_price(2).build().execute_with(|| {
+fn call_doesnt_pay_for_gas() {
```

Comment:
ideally `contracts::Call()` should have some fixed weight that does actually incur some basic fee, neh?

Reply:
This ensures for now that this payment is not performed within the `::call` method itself as it used to be. Now, we totally rely that the weight mechanism will take care of it.

However, you are definitely right! I will add weight annotation to the `::call` method.